main.c

```diff
@@ -86,11 +86,15 @@ static ssize_t dev_write(struct file *filep,
                          size_t len,
                          loff_t *offset)
 {
-    size_of_message = 0;
+    size_of_message = len;
     memset(message, 0, sizeof(char) * BUFF_SIZE);
 
-    copy_from_user(message, buffer, BUFF_SIZE);
-    size_of_message = strlen(message);
+    if (size_of_message >= BUFF_SIZE) {
+        printk(KERN_ALERT "Expression too long");
```

Comment:
Or should we declare `size_of_message` as local variable since it is assigned to `strlen(message)` in `dev_read`?

Reply:
I thought that `size_of_message` means the string length get from user here. I'll just use `len` to do the check

Reply:
Yeah, it's more like a local variable  since it always get reassigned before used. We can declare `size_of_message` as a local variable but not in this PR because it comes from your design.